### PR TITLE
[vcpkg] Use FetchContent instead of submodule

### PR DESCRIPTION
**Pull request by m-kuhn**
~Less~ No submodules, more fun

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 801ff9e8faf5f5565ba61d29f98184685662721c: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3432-8eb6c15fd0056ade8783955ca18edaf658f3f9af-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3432-8eb6c15fd0056ade8783955ca18edaf658f3f9af-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3432-8eb6c15fd0056ade8783955ca18edaf658f3f9af-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3432-8eb6c15fd0056ade8783955ca18edaf658f3f9af-x86-android.apk)
</details>

**Comment by nirvn** [comment]
That's really nice, much cleaner than submodule, less friction when manually backporting fixes.

File: vcpkg/triplets/arm-neon-android.cmake [hunk 1]
```diff
@@ -0,0 +1,9 @@
+set(VCPKG_TARGET_ARCHITECTURE arm)
+set(VCPKG_CRT_LINKAGE dynamic)
+set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_CMAKE_SYSTEM_NAME Android)
+set(VCPKG_BUILD_TYPE release)
+set(VCPKG_MAKE_BUILD_TRIPLET "--host=armv7a-linux-androideabi")
```
Comment (nirvn) [inline vcpkg/triplets/arm-neon-android.cmake:6]: Neon is back? I don't see any different between this and arm-android triblet, and I missing something?

Comment (m-kuhn) [reply vcpkg/triplets/arm-neon-android.cmake:6]: It's used (by triplet name) here:
https://github.com/microsoft/vcpkg/blob/master/scripts/toolchains/android.cmake#L17

We should only be using that one and we will need to when we switch to a newer ndk.

Comment (nirvn) [reply vcpkg/triplets/arm-neon-android.cmake:6]: @m-kuhn , K, makes sense.

File: .github/workflows/qt6.yml [hunk 2]
```diff
@@ -94,7 +94,7 @@ jobs:
 
       - name: 🧫 Test
         env:
-          PROJ_LIB: ${{ env.CMAKE_BUILD_DIR }}/vcpkg_installed/x64-osx/share/proj/data
+          PROJ_LIB: ${{ env.CMAKE_BUILD_DIR }}/vcpkg_installed/x64-osx/share
```
Comment (m-kuhn) [inline .github/workflows/qt6.yml:97]: /proj

**Review by nirvn — APPROVED** [review]

**Pull request merged**